### PR TITLE
Edit "What the compiler does to your code"

### PR DESCRIPTION
**Pull request by pierwill**
Re-writes this section in cohesive paragraphs and sections.

For reviewers: I know the diff is massive, but I promise this is faithful to what was there before. 😇

### CODE REVIEW

File: src/overview.md [hunk 1]
```diff
@@ -17,94 +17,122 @@ So first, let's look at what the compiler does to your code. For now, we will
...
+methods found in [`rustc_parse::parser::Parser`]. The external module parsing
+entry point is [`rustc_expand::module::parse_external_mod`][parse_external_mod].
+And the macro parser entry point is [`Parser::parse_nonterminal()`][parse_nonterminal].
+
+Parsing is performed with a set of `Parser` utility methods including `bump`,
+`check`, `eat`, `expect`, `look_ahead`.
+
+Parsing is organized by the semantic construct that is being parsed. Separate
+`parse_*` methods can be found in [`rustc_parse` `parser`][rustc_parse_parser_dir]
+directory. The source file name follows the construct name. For example, the
```
Comment (camelid) [inline src/overview.md:65]: The wording around `rustc_parse` seems a bit awkward.

File: src/overview.md [hunk 2]
```diff
@@ -17,94 +17,122 @@ So first, let's look at what the compiler does to your code. For now, we will
...
+The token stream passes through a higher-level lexer located in
+[`rustc_parse`] to prepare for the next stage of the compile process. The
+[`StringReader`] struct is used at this stage to perform a set of validations
+and turn strings into interned symbols (_interning_ is discussed later).
+[String interning] is a way of storing only one immutable
+copy of each distinct string value.
+
+The lexer has a small interface and doesn't depend directly on the
+diagnostic infrastructure in `rustc`. Instead it provides diagnostics as plain
+data which are emitted in `rustc_parse::lexer::mod` as real diagnostics.
```
Comment (camelid) [inline src/overview.md:49]: What is `mod` here?

**Review by camelid — COMMENTED** [review]
I don't think I'll have time to review this PR much, but I noticed a couple small things.

**Comment by camelid** [comment]
Can you move the `Ty` link-fix to a new PR?

**Comment by pierwill** [comment]
> Can you move the `Ty` link-fix to a new PR?

Sure. One sec.

**Comment by pierwill** [comment]
I'll try splitting these changes into one commit per paragraph.

File: src/overview.md (hunk 2 above)
Comment (pierwill) [reply src/overview.md:49]: 🤷

Comment (camelid) [reply src/overview.md:49]: Oh, I'd thought you added that. Can you remove the `::mod` suffix?

**Comment by spastorino** [comment]
@pierwill I'd like to merge this, is it ready for review? is there something else missing?

**Comment by pierwill** [comment]
@spastorino I think this is ready to merge. 👍

**Pull request merged**